### PR TITLE
Fix linux compilation

### PR DESCRIPTION
**Pull request by Saarg**
Hi, while playing with imgui on Linux today I wanted to add netimgui in my project and had to do a couple edits to compile.

First I had to add a missing include and fix what I think is a typo in NetImgui_NetworkPosix.cpp but I also had some issues with ImTexture. I think the condition in the assert is doing the opposite of what the message says and I also had to change some reinterpret_cast to static_cast.

Let me know if you think the reinterpret_cast issue could be from my imgui integration and I'll revert the change.

### CODE REVIEW

File: Code/Client/Private/NetImgui_Api.cpp [hunk 1]
```diff
@@ -282,7 +282,7 @@ void SendDataTexture(ImTextureID textureId, void* pData, uint16_t width, uint16_
 
 	// Makes sure even 32bits ImTextureID value are received properly as 64bits
 	uint64_t texId64(0);
-	static_assert(sizeof(uint64_t) <= sizeof(textureId), "ImTextureID is bigger than 64bits, CmdTexture::mTextureId needs to be updated to support it");
+	static_assert(sizeof(uint64_t) >= sizeof(textureId), "ImTextureID is bigger than 64bits, CmdTexture::mTextureId needs to be updated to support it");
```
Comment (sammyfreg) [inline Code/Client/Private/NetImgui_Api.cpp:285]: yep, that was a mistake

File: Code/Client/Private/NetImgui_NetworkPosix.cpp [hunk 2]
```diff
@@ -81,8 +82,8 @@ SocketInfo* ListenStart(uint32_t ListenPort)
 	if( ListenSocket != -1 )
 	{
 		int flag = 1;  
-		setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, &flag, sizeof(flag));
-		setsockopt(sockfd, SOL_SOCKET, SO_REUSEPORT, &flag, sizeof(flag));
+		setsockopt(ListenSocket, SOL_SOCKET, SO_REUSEADDR, &flag, sizeof(flag));
```
Comment (sammyfreg) [inline Code/Client/Private/NetImgui_NetworkPosix.cpp:85]: Yes, this looks like a renaming error occurs.

**Comment by sammyfreg** [comment]
Thanks you for the pull request. I do not gave an environment to test linux build, so I have been relying on people like you to see fi it works well.

**Comment by sammyfreg** [comment]
I have a potential fix for the reinterpret_cast error on linux.
Would you mind removing your static_cast change, and try to pull into the dev branch instead?

Merci.

**Comment by Saarg** [comment]
Salut,

I also added one call to TextureCastHelper in NetImgui_CmdPackets_DrawFrame.cpp and everything compiles fine 👍

**Review by sammyfreg — APPROVED** [review]

**Comment by sammyfreg** [comment]
Hope you find this library useful.

Merci pour le fix linux.

**Pull request merged**